vignettes/documentation_and_dictionaries.Rmd

```diff
@@ -0,0 +1,228 @@
+SOURCE_GROUP_YEARS_DETAILS
+:::
+
+::: {style="text-align: justify;"}
+Demographic datasets are available for municipalities and departments, and contain data for Dwellings, Households, Population and Population Projections in five categories.
+
+-   Viviendas (Dwellings)
+-   Hogares (Households)
+-   Personas Social (Persons social)
+-   Personas Demográfico (Persons demographic)
+
+All datasets are retrieved from the National Administrative Department of Statistics (DANE). Naming is stated as follows:
+
+-   Source: **DANE**
+-   Group: Names include the categories
+    -   Viviendas: **CNPVV**
+    -   Hogares: **CNPVH**
+    -   Personas Social: **CNPVPS**
+    -   Personas Demográfico: **CNPVPD**
+-   Year:
+    -   Census data: **2018**
+-   Details: These are related to each individual dataset. For further details please check the function `list_datasets` below.
+:::
+
+::: {style="text-align: justify;"}
+For hands on examples please check [A Deep Dive into Colombian Demographics Using ColOpenData](https://epiverse-trace.github.io/ColOpenData/articles/demographic_data.html).
+:::
+
+#### Understanding Geospatial Datasets
+
+::: {style="text-align: justify;"}
+Geospatial datasets naming is related to the level of aggregation, since they are available from Blocks to Departments. All these datasets come from DANE, and are part of the Geostatistical National Framework (MGN), which for 2018 included a summarized version of the National Population and Dwelling Census (CNPV). Available spatial levels include: department, municipality, urban and rural sector, urban and rural section, urban zone and blocks. Please check [Maps and plots with ColOpenData](https://epiverse-trace.github.io/ColOpenData/articles/geospatial_data.html) for further details.
+:::
+
+## Understanding Climate Dataset
+
+::: {style="text-align: justify;"}
+This module's data is stored in an unique dataset, and the information required to use the related functions is the area of interest, dates, and tags be consulted. Individual tags are required to download data and include:
+:::
+
+```{r IDEAM table, echo = FALSE}
+tags <- c(
+  "TSSM_CON", "THSM_CON", "TMN_CON", "TMX_CON", "TSTG_CON", "HR_CAL",
+  "HRHG_CON", "TV_CAL", "TPR_CAL", "PTPM_CON", "PTPG_CON", "EVTE_CON",
+  "FA_CON", "NB_CON", "RCAM_CON", "BSHG_CON", "VVAG_CON", "DVAG_CON",
+  "VVMXAG_CON", "DVMXAG_CON"
+)
+variable <- c(
+  "Dry-bulb Temperature", "Wet-bulb Temperature",
+  "Minimum Temperature", "Maximum Temperature",
+  "Dry-bulb Temperature (Termograph)", "Relative Humidity",
+  "Relative Humidity (Hydrograph)", "Vapour Pressure", "Dew Point",
+  "Precipitation (Daily)", "Precipitation (Hourly)", "Evaporation",
+  "Atmospheric Phenomenon", "Cloudiness", "Wind Trajectory",
+  "Sunshine Duration", "Wind Speed", "Wind Direction",
+  "Maximum Wind Speed", "Maximum Wind Direction"
+)
+
+IDEAM_tags <- data.frame(
+  Tags = tags, Variable = variable,
+  stringsAsFactors = FALSE
+)
+knitr::kable(IDEAM_tags)
+```
```

Comment:
> I would make this an exported dataset. Potentially with columns for different language options.
> 
> I see this request is partially covered by `dictionary("IDEAM_CLIMATE_2023_MAY")` but is it necessary to make a request for this? I expect it to be used very regularly and it may make sense to store it locally.
> 
> Do you expect it to be updated frequently, which may justify to keep it online?

We decided to leave them in online just for consistency, so all datasets are retrieved in the same way. Also, the request is not very demanding for these light datasets.

Reply:
> Would there be a real benefit in keeping these datasets (like dictionaries) inside the package?

I think so. I see them as a form of documentation, which should as much as possible be available offline.

Given the size of ColOpenData datasets, it's likely users will save them on their local machine, and maybe work with them in a moment where they don't have internet. It'd be good to ensure they have everything they need to still understand the data.

> We decided to leave them in online just for consistency

I'm not sure there is a need for consistency here as I see two very distinct types of data:

- the actual data
- the dictionaries / documentation

It's fair to treat them differently in my opinion.

Reply:
Now dictionaries are stored locally, those regarding geospatial and climate tags. This was presented in PR #145